### PR TITLE
improve dbkvs config

### PR DESCRIPTION
**Pull request by rjullman**

### CODE REVIEW

**Comment by rjullman** [comment]
This is a pr for organization.  Currently just ignore it (I made it a PR so I can reference it from the other monolith commit).  I'll change the status from "do not merge" when there is content worth looking at here.

**Comment by clockfort** [comment]
looks good so far, seems fine to go in if you want to merge in manageable chunks.

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java [hunk 1]
```diff
@@ -59,36 +58,35 @@
...
+    private final SqlConnectionSupplier connections;
 
-    public DbKvs(ExecutorService executor,
+    public DbKvs(DbKeyValueServiceConfiguration config,
                  DbTableFactory dbTables,
-                 PalantirSqlConnectionSupplier connections,
-                 JdbcHandler jdbcHandler) {
-        super(executor);
+                 SqlConnectionSupplier connections) {
+        super(AbstractKeyValueService.createFixedThreadPool("Atlas Relational KVS", config.poolSize()));
```
Comment (jboreiko) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java]: Any reason we would want to be able to supply this externally, as is done originally?

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/DbKeyValueServiceConfiguration.java [hunk 2]
```diff
@@ -31,23 +31,28 @@
...
 
     @Value.Default
-    public boolean oracleEnableEeFeatures() {
+    public boolean enableOracleEnterpriseFeatures() {
         return false;
     }
 
     @Value.Default
-    public int getRangeConcurrency() {
+    public int poolSize() {
```
Comment (jboreiko) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/DbKeyValueServiceConfiguration.java]: Not sure that poolSize is a good config value name, maybe something a little more explicit.  Maybe databaseConnectionPoolSize?

**Pull request merged**